examples/autosuggest.amp.html

```diff
@@ -53,19 +53,48 @@
   amp-selector .select-option.no-outline.no-outline[option] {
     outline: none;
   }
-  .select-option:focus {
+  .active, .select-option:focus {
     box-shadow: inset rgb(94, 158, 215) 0px 0px 45px;
   }
   .select-option:focus {
     background: #fff;
   }
   .select-option:nth-child(2n) {
-    background: #eee;
+      background: #eee;
   }
+  .dn { display: none; }
   </style>
 </head>
 <body>
+<amp-state id="data"
+           src="/form/autosuggest/query"
+           [src]="'/form/autosuggest/query?q=' + (autosuggest || '')">
+</amp-state>
   <h1>Programming language search</h1>
+<div class="search-container">
+    <input
+            id="query"
+            name="query"
+            type="text"
+            class="search-box"
+            on="input-debounced:AMP.setState({
+            query: event.value,
+            autosuggest: event.value
+          }),
+          autosuggest-list.show,
+          results.hide;
+          change: AMP.setState({
+            query: index >=0 ? data.items[0].results[index].name : event.value,
+            autosuggest: index >=0 ? data.items[0].results[index].name : event.value
+          }),search-form.submit;
+          keydown:AMP.setState({
```

Comment:
https://github.com/ampproject/amphtml/issues/10947 would probably be a cleaner way to implement this. We could potentially do something like this: 
```
    this.registerAction('navigate', this.navigationKeyDownHandler_.bind(this));
```

and then do `keydown:my-selector-id.navigate` here, and I believe you wouldn't need all the index changes.